enclave/src/rpc/polkadex_rpc_gateway.rs

```diff
@@ -102,8 +103,17 @@ impl RpcGateway for PolkadexRpcGateway {
         proxy_account: Option<AccountId>,
         nonce: u32,
     ) -> Result<(), GatewayError> {
-        authenticate_user_and_validate_nonce(main_account, proxy_account, nonce)
+        let result =
+            authenticate_user_and_validate_nonce(main_account.clone(), proxy_account, nonce);
+        if result.is_ok() {
```

Comment:
Hmm... with our benchmarking feature this will not be part of benchmarking at all.. is that actually okay? @Gauthamastro   @zktony

Reply:
Benchmarking feature was a temporary solution. It will not be used for production benchmarking when everything is done. Also it will be removed soon.

Reply:
Can this be marked as resolved, then?

Reply:
I think so.